pkg/cgroup/resolve_container_test.go

```diff
@@ -121,3 +122,40 @@ func TestGetAliveContainers(t *testing.T) {
 		})
 	}
 }
+func TestExtractPodContainerIDfromPathWithCgroup(t *testing.T) {
+	tests := []struct {
+		name           string
+		path           string
+		cgroup         int
+		expectedResult string
+	}{
+		{
+			name:           "valid path with cgroup 1",
+			path:           "1:name=systemd:/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-podd0511cd2_29d2_4215_be0f_f77bc0609d99.slice/crio-f93ee491b8ed2680d5a909eb098b14a9430173b57ca1c4efedd8768566d67e8e.scope",
+			cgroup:         1,
+			expectedResult: "f93ee491b8ed2680d5a909eb098b14a9430173b57ca1c4efedd8768566d67e8e",
+		},
+		{
+			name:           "valid path with cgroup 2",
+			path:           "0::/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod2c9f8a79_5391_454b_88cb_86190881cb96.slice/crio-a09343ca97901516c25036e2b954421254f8c68b384b536064e8999f0c4ed18d.scope",
+			cgroup:         2,
+			expectedResult: "a09343ca97901516c25036e2b954421254f8c68b384b536064e8999f0c4ed18d",
+		},
+		{
+			name:           "valid path with cgroup v1 with container",
+			path:           "7:cpu,cpuacct:/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-podd0511cd2_29d2_4215_be0f_f77bc0609d99.slice/crio-f93ee491b8ed2680d5a909eb098b14a9430173b57ca1c4efedd8768566d67e8e.scope/container",
+			cgroup:         1,
+			expectedResult: utils.SystemProcessName,
+		},
+		// Add more test cases as needed
```

Comment:
Can you add another test with the following path from: https://github.com/sustainable-computing-io/kepler/issues/923#issuecomment-1759202284
```
13:hugetlb:/system.slice/
docker-2fa3e04b676df750842faf017052dd37ea0cc5bc7259a3487a1718c7fe100c94.scope/
kubelet/kubepods/besteffort/
podbdd4097d-6795-404e-9bd8-6a1383386198/
c79788e0da15a6597263eb2b9c51d05dd1a9a1d08c53c1161dc8c45d2dac6b38
```
The container id is `c79788e0da15a6597263eb2b9c51d05dd1a9a1d08c53c1161dc8c45d2dac6b38`

Reply:
Also, could please add the case from https://github.com/sustainable-computing-io/kepler/issues/107#issuecomment-1216908476
```
/sys/fs/cgroup/systemd/system.slice/containerd.service/kubepods-burstable-poda3b200c9_db51_40b4_9d2d_53f8fdf80d7f.slice:cri-containerd:286b15051ec43375190802e1f40562536980a8fd97e75bb89c7f2eec6f995f17
```
The container id is: `286b15051ec43375190802e1f40562536980a8fd97e75bb89c7f2eec6f995f17`

Reply:
And finally the case PR #684

```
11:blkio:/kubepods/burstable/podf6adb0af-0855-4bab-b25b-c853f18d0ce2/35b97177dada20362ab90d90ac63cd54e8a41cf87bea34f270631b6da17f4a93
```
The container id is: `35b97177dada20362ab90d90ac63cd54e8a41cf87bea34f270631b6da17f4a93`

Reply:
So, four new cases to be tested :)

Reply:
@rootfs could you please add those use cases?

Reply:
added